playbooks/operationalizing/secure-forward-splunk.adoc

```diff
@@ -0,0 +1,318 @@
+  private_key_passphrase ocpsecureforward <5>
+</source>
+
+
+<match **>
+   type splunk_ex
+   host 10.9.49.71 <6>
+   port 9997 <7>
+   output_format json <8>
+</match>
+----
+<1> Port to accept incoming logs
+<2> A shared value between the sender and the receiver
+<3> Location of the previously generated certificate
+<4> Location of the previously generated private key
+<5> Private key passphrase
+<6> Hostname or IP of the Splunk instance
+<7> Port number of the Splunk input configured to accept messages
+<8> Format in which messages are sent to Splunk
+
+
+NOTE: To allow messages to be received on port 24284 for the secure forward plugin, ensure the proper firewall configurations are in place
+
+Restart the _td-agent_ service to apply the changes
+
+[source]
+----
+systemctl restart td-agent
+----
+
+=== OpenShift Aggregated Logging
+
+Once Splunk and the standalone instance of Fluentd have been configured, OpenShift's aggregated logging framework can be configured to securely forward messages externally.
+
+
+==== Configure Certificates
+
+As previously configured, certificates were generated and implemented in the standalone Fluentd instance to provide secure communication for the secure forward plugin between Fluentd running on each node in OpenShift and the standalone Fluentd instance. The same certificates need to now be configured in OpenShift.
+
+Copy the files from the _/etc/td-agent/certs/_ folder to a location on your local machine. 
+
+Login to the OpenShift environment as a user with privileges to modify the logging infrastructure and change to the _logging_ project:
+
+[source]
+----
+oc login <openshift_master_address>
+oc project logging
+----
+
+A secret called _logging-fluentd_ are already configured within the logging infrastructure in order to communicate with elastic search. Patch the existing secret to include the certicate and private key copied from the standalone Fluentd instance.
+
+[source]
+----
+oc patch secrets/logging-fluentd --type=json --patch "[{'op':'add','path':'/data/external_ca_cert.pem','value':'$(base64 /path/to/your_ca_cert.pem)'}]"
+oc patch secrets/logging-fluentd --type=json --patch "[{'op':'add','path':'/data/external_ca_key.pem','value':'$(base64 /path/to/your_private_key.pem)'}]"
+----
+
+[NOTE]
+====
+The names external_ca_cert.pem and external_ca_key.pem provided in the code sample are the key names configured within the patched secret
+
+The base64 utility must be present on the machine executing the commands
+====
+
+
+==== Configuring Fluentd
+
+A link:https://docs.openshift.com/container-platform/latest/dev_guide/configmaps.html[ConfigMap] called _logging-fluentd_ is configured within the aggregated logging framework to specify the values to containing the configurations for Fluentd. Multiple files are containing within the ConfigMap, including a file called _secure-forward.conf_. The contents of this file are commented out as secure forward plugin is not enabled by default.
```

Comment:
in first sentence "...specify the values to containing the configurations for Fluentd".  I think removing the work 'to' makes the sentence read better.

And in the second sentence, update 'containing' to 'contained':  "Multiple files are **contained** within..."